src/components/ComputeModal.js

```diff
@@ -545,19 +545,24 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
         existingRuntime.masterMachineType !== desiredRuntime.masterMachineType)
   }
 
-  const makeImageInfo = style => div({ style: { whiteSpace: 'pre', ...style } }, [
-    div({ style: Style.proportionalNumbers }, ['Updated: ', updated ? Utils.makeStandardDate(updated) : null]),
-    h(Link, {
-      href: getChangelogUrl(_.find({ image: selectedLeoImage }, leoImages)),
-      disabled: !isTerraSupported(_.find({ image: selectedLeoImage }, leoImages)),
-      ...Utils.newTabLinkProps
-    }, ['Version: ', version || null]),
-    h(ClipboardButton, {
-      text: selectedLeoImage,
-      style: { marginLeft: '0.5rem' },
-      tooltip: 'Copy the image version'
-    })
-  ])
+  const makeImageInfo = style => {
+    const isTerraSupported = ({ isCommunity = false, id }) => !isCommunity && !(getToolForImage(id) === tools.RStudio.label)
+    const getChangelogUrl = ({ id }) => `https://github.com/DataBiosphere/terra-docker/blob/master/${_.replace('_legacy', '', id)}/CHANGELOG.md`
+
+    return div({ style: { whiteSpace: 'pre', ...style } }, [
+      div({ style: Style.proportionalNumbers }, ['Updated: ', updated ? Utils.makeStandardDate(updated) : null]),
+      h(Link, {
+        href: leoImages.length > 0 ? getChangelogUrl(_.find({ image: selectedLeoImage }, leoImages)) : '',
+        disabled: leoImages.length > 0 ? !isTerraSupported(_.find({ image: selectedLeoImage }, leoImages)) : true,
+        ...Utils.newTabLinkProps
+      }, ['Version: ', version || null]),
+      h(ClipboardButton, {
+        text: selectedLeoImage,
+        style: { marginLeft: '0.5rem' },
+        tooltip: 'Copy the image version'
+      })
+    ])
+  }
```

Comment:
Consider 

```suggestion
  const makeImageInfo = style => {
    const selectedImage = _.find({ image: selectedLeoImage }, leoImages)
    const shouldDisable = _.isEmpty(leoImages) ? true : selectedImage.isCommunity || getToolForImage(selectedImage.id) === tools.RStudio.label
    const changelogUrl = _.isEmpty(leoImages) ?
      '' :
      `https://github.com/DataBiosphere/terra-docker/blob/master/${_.replace('_legacy', '', selectedImage.id)}/CHANGELOG.md`

    return div({ style: { whiteSpace: 'pre', ...style } }, [
      div({ style: Style.proportionalNumbers }, ['Updated: ', updated ? Utils.makeStandardDate(updated) : null]),
      h(Link, {
        href: changelogUrl,
        disabled: shouldDisable,
        ...Utils.newTabLinkProps
      }, ['Version: ', version || null]),
      h(ClipboardButton, {
        text: selectedLeoImage,
        style: { marginLeft: '0.5rem' },
        tooltip: 'Copy the image version'
      })
    ])
  }
```

mainly since `isTerraSupported` and `getChangelogUrl` no longer need to be functions.

Reply:
Thank you! Those changes make a lot of sense to me